### PR TITLE
Allow viewing of Tor browser functional test over VNC

### LINKED ISSUE #3718: [functional testing] Add VNC server to functional test runner container
## Description

It is very difficult currently to debug why functional tests fail using the new tor browser selenium logic. The upstream selenium team offers docker containers packed with firefox/chrome that include a VNC server for visual view of what the browser is doing. This would be insanely useful for us during development.

## User Research Evidence

Doesn't affect end-users, only developers.

## User Stories

As a developer, it would aid in my troubleshooting to be able to visually see the Tor Browser tests running while they are running.

Epic: #3659

### PR DESCRIPTION
**Pull request by msheiny**
## Status

Ready for review

## Description of Changes

Fixes #3718 

Changes proposed in this pull request:
* Does some re-configuration with the existing VNC helper function and uses Xvnc instead (quite frankly I had a lot of issues trying to get it working with `x11vnc` + `Xvfb` )
* Adds some helper commands 

## Testing

How should the reviewer test this PR?

If you want to test against a remote server:
* Start a staging server (either via the upgrade scenario or manually)
* If you didn't use `upgrade` scenario - fill out the `instance_information.json` and do the create_dev dance
* Start the tests `cd securedrop ;  ./bin/dev-shell ./bin/run-test tests/functional` 
* While those are running (once you start seeing test section) run `make func-vnc` )
* :tada:  you should see the tests running 

Of special note is that I did `NOT` run this under macOS yet so I'll need anyone with `macOS` *cough* *cough* @redshiftzero *cough* to take a look and make sure that `make func-vnc` call works

## Checklist

- [ ] How many woodchucks?

### CODE REVIEW

**Comment by msheiny** [comment]
oh i need a rebase here.. hold on

**Comment by msheiny** [comment]
ummmmmm i have no idea why that admin test is failing... waiting until the rest finish to restart but it passes locally with no probs

File: devops/scripts/vnc-docker-connect.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+set -e
+
+# Bomb out if container not running
+docker inspect securedrop-dev >/dev/null 2>&1 || (echo "ERROR: SD container not running."; exit 1)
+
+
+# Find our securedrop docker ip
+SD_DOCKER_IP="$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' securedrop-dev)"
+
+# Quit if the VNC port not found
```
[the remark this answers is not in the document]
Comment (msheiny) [reply devops/scripts/vnc-docker-connect.sh]: hey @redshiftzero - Yeah, I agree but I don't want this to be another feature that doesn't have Mac support. :-1:  Let me see if I can tweak some things real quick

**Comment by zenmonkeykstop** [comment]
Got VNC working with functional tests on OS X by:
1) Adding -P flag to docker run command to publish ports
2) using `docker port securedrop-dev` to find what 5901 was mapped to locally (could skip this by using -p DOCKERPORT:LOCALPORT instead of -P)
3) starting up a vnc client (tigerVNC in my case, built-in VNC wasn't working for me) pointing at 0.0.0.0:LOCALPORT while the tests were running.

So it's absolutely possible for OS X, if you're ok with publishing the ports from the securedrop-dev container.

**Comment by redshiftzero** [comment]
superseded by #3817

**Pull request closed without merging**